### DISCUSSION TITLE
Artifactory folder name

### DISCUSSION THREAD

**Discussion by PatchaIT** [community/community#143571] [New to GitHub] [answered]
Hi all,
I'm pretty new here.
I created a project with some python and other files. It's private because I'm just familiarizing myself with the tool.

I tried to generate a release, and I noticed that it uses the project name both for zip file and for first folder inside the zip file.

Seeing that i want to adopt a prefix into project name, to make clear for what the script is, but I don't want that prefix to come out into the artifact filename. Or at least to customized first folder inside the artifact.

How to do that?

The project name I want to use is slcb-greetings, while I wish the artifact to just be Greetings.zip or at least the first folder inside the zip to be named just "Greetings".

Any way to do that?
Thank you

**Comment by techy4shri** [comment, marked as answer]
You can control the ZIP file name by modifying the GitHub Actions workflow so that the ZIP file generated is named Greetings.zip.
Then, you can rename the folder inside the zip file (from slcb-greetings to Greetings), you will need to:
1. Temporarily rename the folder to Greetings.
2. Zip it.
3. Rename the folder back to slcb-greetings.

Afterwards, follow these steps:
1. Navigate to the Releases section of your repository.
2. Create a new release, to trigger the workflow.

Hope this help!

**Reply by PatchaIT** [reply]
> You can control the ZIP file name by modifying the GitHub Actions workflow

I don't get if I should expect to alraedy have a workflow to modify (which I have none) or I have to create one.

I'm having difficulties to make or find a workflow which edits release asset files.
Maybe you adopted a release-maker workflow by yourself and you're expecting everybody has one?

In case, which one  you've got?

**Reply by techy4shri** [reply]
Go to github actions workflows and you will find the workflow you need to modify there.

**Reply by PatchaIT** [reply]
> Go to github actions workflows and you will find the workflow you need to modify there.

It's empty.
There's no workflow at the start.

Probably you added your own time ago.

**Reply by techy4shri** [reply]
![image](https://github.com/user-attachments/assets/0b46365c-479f-4a7f-8911-3374f4de526a)
![image](https://github.com/user-attachments/assets/c562a1cd-872b-4791-9140-34fece9b587c)
Here!

**Reply by PatchaIT** [reply]
I think there's a misunderstanding.
That is just a namesake workflow template which greets users when they do first issue or first pull-request.
It's not related with my script (which is a python plugin script for Streamlabs Chatbot) and it's not related with releases and assets.

Anyway up to now I'm trying by using a workflow which uses an action that should take a zip file created by bash command, and put it between my assets into a release.
I still didn't succeed to make it work, but I'm working on it.
The action it's this one:
https://github.com/marketplace/actions/upload-release-asset

So I still hope Github Actions will be my solution. :)

**Reply by PatchaIT** [reply]
Actually I resolved with a workflow, but not using that action (it is deprecated right now).
I solved with the workflow I "answered myself" into this other discussion:

https://github.com/orgs/community/discussions/144232

**Reply by techy4shri** [reply]
You do have to create a workflow suited for you, I just guided you in getting started, can't give the full code now for that, can I? :)
Glad you worked it out!

**Comment by PatchaIT** [comment]
Thank you for the replay.
So I have to check GitHub Actions.

I'm making some changes now (on my spare time), I'll check this instruction before next time I'll try to make a new release after those changes. ;)

**Reply by techy4shri** [reply]
Sure! Happy to help!

**Reply by ghostinhershell** [reply]
Hi @PatchaIT 👋🏾,I'm not sure if you've had the time to test this out on your end but, if @techy4shri's answer helped you here, can you please mark their response as the answer? When you mark a question as an answer, GitHub will highlight the comment and replies to the comment to help visitors quickly find the answer.
![image](https://github.com/user-attachments/assets/b89e7161-c2d1-47e2-a058-242df633e6b2)

If it doesn't, please feel free to let us know so that we can take another look here. Thanks!

**Reply by PatchaIT** [reply]
> Hi @PatchaIT 👋🏾,I'm not sure if you've had the time to test this out on your end but, if @techy4shri's answer helped you here, can you please mark their response as the answer? When you mark a question as an answer, GitHub will highlight the comment and replies to the comment to help visitors quickly find the answer. ![image](https://private-user-images.githubusercontent.com/14021469/353319770-b89e7161-c2d1-47e2-a058-242df633e6b2.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzA5MTU0MDAsIm5iZiI6MTczMDkxNTEwMCwicGF0aCI6Ii8xNDAyMTQ2OS8zNTMzMTk3NzAtYjg5ZTcxNjEtYzJkMS00N2UyLWEwNTgtMjQyZGY2MzNlNmIyLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDExMDYlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQxMTA2VDE3NDUwMFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTJmYzdjY2FkZmUzN2Y1ODc2M2RhNjQ0NWYzMzRjMjBiMzZiMTZiZWE1NTA3MGQwOGE0OGMwODE4MTVlMmVmMGImWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.vRB-TiGR3UioFNSMv7RKkzZf6cjLeIqWHMkiRZnelL0)
> 
> If it doesn't, please feel free to let us know so that we can take another look here. Thanks!

Hi,
I'm sorry, but I only "play" with my github projects when I have spare time.
And this is harder to happen during working days.

Surely as soon as I'll check what suggested, I'll mark the response.
I hope there's no max time to do so, but I suppose it will be within this week.

Maybe later, after work, I could have some spare time, I hope.

**Comment by PatchaIT** [comment]
Uhm... I'm finding that creating or modifing a workflow is something pretty new to me.
So I'll need time to figure it out.

What I can say up to now is that this link doesn't work:
![image](https://github.com/user-attachments/assets/2cbfd139-bb1e-4a3e-b474-41dc14d979ea)

**Comment by PatchaIT** [comment]
I suppose this will help me, and it will take a while. :)
https://docs.github.com/en/actions/writing-workflows/quickstart

So for now I'll trust the answer. ;)

**Comment by PatchaIT** [comment]
It's empty.
There's no workflow at the start.

Probably you added your own time ago.

________________________________
Da: Garima ***@***.***>
Inviato: sabato 9 novembre 2024 20:59
A: community/community ***@***.***>
Cc: Patcha ***@***.***>; State change ***@***.***>
Oggetto: Re: [community/community] Artifactory folder name (Discussion #143571)


Go to github actions workflows and you will find the workflow you need to modify there.

**Comment by techy4shri** [comment]
@PatchaIT I just got a notification that my answer was unmarked...may I know the reason behind this please?

**Reply by PatchaIT** [reply]
I didn't do that.

Don't know if somebody else could do that.

**Reply by PatchaIT** [reply]
Infact, I don't know why, now the topic results unanswered, even if I solved.

**Comment by techy4shri** [comment]
@ghostinhershell hey can you check who is this ghost account? it says a deleted account but how could it un-mark an answer?

**Reply by ghostinhershell** [reply]
Hey @techy4shri, 

When a GitHub account is deleted, their resources and comments will become associated with the [ghost user](https://github.com/ghost). So I won't be able to pinpoint exactly which user unmarked the answer here before their account was deleted. 

@PatchaIT, can you please mark the correct response as the answer here again? If you continue to run into issues doing this, I'll escalate this to our support team.

**Reply by PatchaIT** [reply]
Ok did

**Reply by techy4shri** [reply]
Thanks!
